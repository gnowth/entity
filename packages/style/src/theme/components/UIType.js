import { css } from 'styled-components';

export const component_uiType_h1 = {
  as: 'h1',
  css: css`
    font-size: 6rem;
    font-weight: 300;
    letter-spacing: -1.5px;
  `,
};

export const component_uiType_h2 = {
  as: 'h2',
  css: css`
    font-size: 3.75rem;
    font-weight: 300;
    letter-spacing: -0.5px;
  `,
};

export const component_uiType_h3 = {
  as: 'h3',
  css: css`
    font-size: 3rem;
    font-weight: 400;
    letter-spacing: 0;
  `,
};

export const component_uiType_h4 = {
  as: 'h4',
  css: css`
    font-size: 2.125rem;
    font-weight: 400;
    letter-spacing: 0.25px;
  `,
};

export const component_uiType_h5 = {
  as: 'h5',
  css: css`
    font-size: 1.5rem;
    font-weight: 400;
    letter-spacing: 0;
  `,
};

export const component_uiType_h6 = {
  as: 'h6',
  css: css`
    font-size: 1.25rem;
    font-weight: 500;
    letter-spacing: 0.15px;
  `,
};

export const component_uiType_subtitle1 = {
  as: 'p',
  css: css`
    font-size: 1rem;
    font-weight: 400;
    letter-spacing: 0.15px;
  `,
};

export const component_uiType_subtitle2 = {
  as: 'p',
  css: css`
    font-size: 0.875rem;
    font-weight: 500;
    letter-spacing: 0.1px;
  `,
};

export const component_uiType_body1 = {
  as: 'p',
  css: css`
    font-size: 1rem;
    font-weight: 400;
    letter-spacing: 0.5px;
  `,
};

export const component_uiType_body2 = {
  as: 'p',
  css: css`
    font-size: 0.875rem;
    font-weight: 400;
    letter-spacing: 0.25px;
  `,
};

export const component_uiType_button = {
  as: 'span',
  css: css`
    font-size: 0.875rem;
    font-weight: 500;
    letter-spacing: 1.25px;
    text-transform: uppercase;
  `,
};

export const component_uiType_caption = {
  as: 'p',
  css: css`
    font-size: 0.75rem;
    font-weight: 400;
    letter-spacing: 0.4px;
  `,
};

export const component_uiType_label = {
  as: 'label',
  css: css`
    font-size: 0.875rem;
    font-weight: 400;
    letter-spacing: 0.25px;
  `,
};

export const component_uiType_overline = {
  as: 'p',
  css: css`
    font-size: 0.625rem;
    font-weight: 400;
    letter-spacing: 1.5px;
    text-transform: uppercase;
  `,
};
