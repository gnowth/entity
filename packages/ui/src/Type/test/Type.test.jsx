import 'jest-styled-components';
import renderer from 'react-test-renderer';
import React from 'react';
import { IntlProvider } from 'react-intl';

import UIType from '..';

it('renders correctly', () => {
  const tree = renderer.create(
    <IntlProvider locale="en" messages={{}}>
      <UIType>Test</UIType>
    </IntlProvider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
