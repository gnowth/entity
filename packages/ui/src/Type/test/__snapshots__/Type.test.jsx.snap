// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
@media print {
  .c0 {
    color: black;
  }
}

<span
  className="c0"
>
  Test
</span>
`;
